### PR TITLE
Add rubocop linter and fix issues

### PR DESCRIPTION
**Pull request by potatoqualitee**
Note: I do not know Ruby and these are just proposals.

This syntax checker (7de9e4ce06e89092570305741b4e2786397cad5d) will run each time that a commit is pushed to the repo, as well as when a Pull Request is issued in order to prevent syntax issues that were detailed in https://github.com/vmware/dod-compliance-and-automation/issues/78

1. I added syntax checking via Chef's `cookstyle`
1. I made it autofix a bunch of stuff (`cookstyle -A ./`)
1. I fixed what I could by guessing in commit 65be31540acbc10836763abb1ca482204225b71f

I don't do regex so I disabled the check for `Lint/ImplicitStringConcatenation` in `.rubocop.yml` instead of attempting to fix it. But here is the failure in case anyone is interested. If this is fixed, please remove the `Lint/ImplicitStringConcatenation` section in  `.rubocop.yml`.

```
vsphere/7.0/vcsa/inspec/vmware-vcsa-7.0-stig-baseline/vami/controls/VCLD-70-000019.rb:63:20: W: Lint/ImplicitStringConcatenation: Combine "/opt/vmware/sbin/vami-lighttpd -p -f /opt/vmware/etc/lighttpd/lighttpd.conf 2>/dev/null|awk '/mimetype\.assign/,/\)/'|grep \"text/\"|grep -v " and 'charset=utf-8' into a single string literal, rather than using implicit string concatenation. Or, if they were intended to be separate method arguments, separate them with a comma.
  describe command("/opt/vmware/sbin/vami-lighttpd -p -f /opt/vmware/etc/lighttpd/lighttpd.conf 2>/dev/null|awk '/mimetype\.assign/,/\)/'|grep \"text/\"|grep -v "'charset=utf-8'"|sed -e 's/^[ ]*//'") do
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
vsphere/7.0/vcsa/inspec/vmware-vcsa-7.0-stig-baseline/vami/controls/VCLD-70-000019.rb:63:163: W: Lint/ImplicitStringConcatenation: Combine 'charset=utf-8' and "|sed -e 's/^[ ]*//'" into a single string literal, rather than using implicit string concatenation. Or, if they were intended to be separate method arguments, separate them with a comma.
  describe command("/opt/vmware/sbin/vami-lighttpd -p -f /opt/vmware/etc/lighttpd/lighttpd.conf 2>/dev/null|awk '/mimetype\.assign/,/\)/'|grep \"text/\"|grep -v "'charset=utf-8'"|sed -e 's/^[ ]*//'") do
```

I also ran the Photon tests before and after (well, from master then tests) and both returned the same results so this appears to work.

```
PS> inspec exec C:\Inspec\Profiles\vmware-photon-3.0-stig-inspec-baseline -t ssh://root@vsphere.ad.local -i $home\.ssh\vsphere --show-progress --reporter=cli json:C:\Inspec\Reports\photon.json
```

Which, both times, resulted in:

```
Profile Summary: 102 successful controls, 15 control failures, 0 controls skipped
Test Summary: 334 successful, 21 failures, 0 skipped

Profile Summary: 102 successful controls, 15 control failures, 0 controls skipped
Test Summary: 334 successful, 21 failures, 0 skipped
```

The `Layout/EndOfLine` exclusion was added because I believe git or GitHub forces incompatible line breaks, so just accept it.

### CODE REVIEW

**Comment by potatoqualitee** [comment]
You can see the successful test workflow run here: https://github.com/potatoqualitee/dod-compliance-and-automation/runs/5551837185?check_suite_focus=true

**Comment by rlakey** [comment]
Going to need to tweak the linting rules...a lot of those changes are not something i would necessarily do. Like removing the space at the end of some lines in the descriptions that's there because the text is split up over multiple lines.

File: ubuntu/18.04/inspec/canonical-ubuntu-18.04-lts-stig-baseline/controls/V-219149.rb [hunk 1]
```diff
@@ -40,15 +38,14 @@
   tag rid: 'SV-219149r508662_rule'
   tag stig_id: 'UBTU-18-010002'
   tag fix_id: 'F-20873r304776_fix'
-  tag cci: ['V-100523', 'SV-109627', 'CCI-001464']
+  tag cci: %w(V-100523 SV-109627 CCI-001464)
```
Comment (rlakey) [inline ubuntu/18.04/inspec/canonical-ubuntu-18.04-lts-stig-baseline/controls/V-219149.rb:41]: It shouldn't be doing this either.

**Comment by potatoqualitee** [comment]
i was thinking about that and wondered. i can make minimal changes by disabling a bunch of rules then you all can enable them slowly as you like.

**Comment by rlakey** [comment]
I added a workflow for the 6.7 stuff which is merged now then we can work on fixing the rest

**Pull request closed without merging**